svm/src/transaction_processor.rs

```diff
@@ -124,16 +124,21 @@ pub struct TransactionProcessingConfig<'a> {
 pub struct TransactionProcessingEnvironment<'a> {
     /// The blockhash to use for the transaction batch.
     pub blockhash: Hash,
+    /// Lamports per signature that corresponds to this blockhash.
+    ///
+    /// Note: This value is primarily used for nonce accounts. If set to zero,
+    /// it will disable transaction fees. However, any non-zero value will not
+    /// change transaction fees. For this reason, it is recommended to use the
+    /// `fee_per_signature` field to adjust transaction fees.
+    pub blockhash_lamports_per_signature: u64,
```

Comment:
is this used for anything other than setting fees to zero?
I'm trying to remove that functionality now in #3211.

Is this exposed to user tx somehow? i.e. can user programs get this from some syscall?